nav2_map_server/src/map_saver.cpp

```diff
@@ -99,9 +99,10 @@ int main(int argc, char ** argv)
   auto map_gen = std::make_shared<nav2_map_server::MapGenerator>(mapname, threshold_occupied,
       threshold_free);
 
+  rclcpp::Rate r(10);
```

Comment:
I think this could suffer from the same deadlock; the node is not spinning while sleep is called.

Reply:
Will do :+1: